core/sys/darwin/Foundation/NSEvent.odin

```diff
@@ -105,6 +105,28 @@ PointingDeviceType :: enum UInteger {
 	Eraser  = 3,
 }
 
+EventModifierFlag :: enum UInteger {
```

Comment:
This type is not used anywhere as far as I can tell, why add it?

Reply:
Because it's a thing? It's not like our Foundation bindings are complete. I plan to keep adding to them progressively. And I plan to use this particular one in the Cocoa/Metal example I will submit to `odin-examples` repo later.

Reply:
What's the end goal for `core/sys/darwin/Foundation/`? To have the bare minimum amount of stuff in there or to have a complete set of bindings that match all features provided by Foundation?

Reply:
I just asked because it doesn't make sense to have a type but have nothing use that type. I see in your snippet that you are transmuting something to it, but that use isn't obvious from reading your PR.

Additionally, could we make it so you don't have to transmute so that type is returned directly? Or is there a reason to leave it like this?

As for the end goal, I believe it is a similar goal as sys/windows where we bind all the useful stuff that people using Odin actually need.

Reply:
Oh, maybe I misunderstood. The type is there to be able to easily transmute `event->modifierFlags()` which is a `UInteger` to Odin's native `bit_set`. It's more ergonomic to have it this way IMO, than to do things like

```
flags := event->modifierFlags()
if (flags & EventModifierFlagCapsLock) ==  EventModifierFlagCapsLock { ... }
```

I suppose I could make `event->modifierFlags()` return the `bit_set`, like with `EventPhase` for example:
```
@(objc_type=Event, objc_name="phase")
Event_phase :: proc "c" (self: ^Event) -> EventPhase {
	return msgSend(EventPhase, self, "phase")
}
```
Which is actually pretty neat, I didn't see this before. Would that be a preferred way?

I see, yeah, as much as I'd prefer to have all the Foundation covered, because it's not possible to know what they might or might not need, I only have the time to create the binding for the things I actually intend to use.

Reply:
Yep making it return the bit set type directly seems like the proper move here

Reply:
Got it, should be fixed now.